include/Audio-player/MusicLibrary.hpp

```diff
@@ -1,49 +1,62 @@
-#ifndef MUSICLIBRARY_MUSICLIBRARY_HPP
-#define MUSICLIBRARY_MUSICLIBRARY_HPP
+#ifndef MUSICLIBRARY_INCLUDE_AUDIO_PLAYER_MUSICLIBRARY_HPP
+#define MUSICLIBRARY_INCLUDE_AUDIO_PLAYER_MUSICLIBRARY_HPP
 
-#include <vector>
-#include <iostream>
-#include <algorithm>
-#include <type_traits>
 #include <filesystem>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
 
+#include "Album.hpp"
+#include "FileManager.hpp"
 #include "Playlist.hpp"
 #include "Song.hpp"
-#include "Album.hpp"
 
 namespace fs = std::filesystem;
 
 class MusicLibrary {
-    std::vector<Playlist> playlists;
-    std::vector<Song> songs;
-    std::vector<Album> albums;
-    fs::path directory;
-
-public:
+  public:
     MusicLibrary() = default;
+
     MusicLibrary(fs::path _directory);
-    void addSong(const Song& song);
-    void addPlaylist(const Playlist& playlist);
 
-    template <typename T>
-    void printVector(const std::vector<T> &vec);
+    void addPlaylist(const Playlist &playlist);
 
-//    template <typename T>
-//    void removeItem(const std::string &itemTitle);
+    template <typename T> void printVector(const std::vector<T> &vec);
 
-    template <typename T>
-    std::vector<T>& getVector();
+    template <typename T> bool isEmpty(std::vector<T> &vector);
 
-    template <typename T>
-    bool isEmpty(std::vector<T> &vector);
+    void updateSongs(FileManager &fm);
+
+    void addSongToAlbum(
+        const std::string &albumName, const std::shared_ptr<Song> &song);
+
+    void updateAlbums();
+
+    const std::unordered_map<std::string, Album> &getAlbumsMap() const {
+        return allAlbumsMap;
+    }
+
+    std::vector<Album> &getAlbums() { return allAlbums; }
+
+    Song getSong(const std::string &songTitle);
+
+    std::vector<std::shared_ptr<Song>> &getSongs() { return allSongs; }
+
+    std::vector<Playlist> &getPlaylists() { return allPlaylists; }
 
     template <typename T>
-    void sortBy(std::vector<T>& vector, bool(*compare)(const T&, const T&)) {
+    inline void
+    sortBy(std::vector<T> &vector, bool (*compare)(const T &, const T &)) {
         std::sort(vector.begin(), vector.end(), compare);
     }
 
-    template <typename T>
-    std::vector<T>& getItem();
+  private:
+    std::vector<Playlist> allPlaylists;
+    std::vector<std::shared_ptr<Song>> allSongs;
```

Comment:
consider 'using x = y' for long names like this that doesn't tell much about what it does. For example 'using SongVector = std::vector<std::shared_ptr<Song>>' and then  SongVector allSongs sound better than std::vector<std::shared_ptr<Song>> allSongs. but you can come up with something better than this name.